flow/project.py

```diff
@@ -280,24 +280,8 @@ def __init__(self, id, name, job, cmd, directives=None):
         # keys set by the user and check whether they have been evaluated by the template
         # script engine later.
         keys_set_by_user = set(directives)
-
-        nranks = directives.get('nranks', 1)
-        nthreads = directives.get('omp_num_threads', 1)
-
-        if callable(nranks) or callable(nthreads):
-            def np_callable(job):
-                nr = nranks(job) if callable(nranks) else nranks
-                nt = nthreads(job) if callable(nthreads) else nthreads
-                return nr*nt
-
-            directives.setdefault('np', np_callable)
-        else:
-            directives.setdefault('np', nranks*nthreads)
-
-        directives.setdefault('ngpu', 0)
-        directives.setdefault('nranks', 0)
-        directives.setdefault('omp_num_threads', 0)
-        directives.setdefault('processor_fraction', 1)
+        if not directives.get('np'):
```

Comment:
This won't be necessary once `_create_run_job_operations` uses `_resolve_directives`.